modules/dnn/src/layer_internals.hpp

```diff
@@ -153,9 +153,10 @@ struct DataLayer : public Layer
         for (int i = 0; i < inputsData.size(); ++i)
         {
             bool isFP16 = outputs[i].depth() == CV_16F;
-            if (inputsData[i].type() == CV_32S || inputsData[i].type() == CV_64S) {
+            if (inputsData[i].type() != CV_32F)
+            {
                 CV_CheckTypeEQ(outputs[i].type(), inputsData[i].type(), "");
-                CV_Assert(means[i] == Scalar() && scaleFactors[i] == 1.0);
+                CV_CheckTrue(means[i] == Scalar() && scaleFactors[i] == 1.0, "Input mean and scale are supported only for float32 input");
```

Comment:
Make it a warning (or info log) instead? It does not crash the program in case users give mean and scale if input is of type other than float (which is rare though) and can have better experience I thought.

Reply:
I don't agree. 
If the program crashes and prints that the input is not supported, the user just easily fixes the input. I think that is a normal experience with a library.
If a user sets mean and scale, but we ignore it, the model returns a wrong result. It is not guaranteed that the user notices the warning message or notices the wrong result, so he will have some problems later that will be hard to find.